blueocean-dashboard/src/main/js/components/Branches.jsx

```diff
@@ -22,30 +23,34 @@ export default class Branches extends Component {
                 location,
                 pipeline: {
                     name: pipelineName,
+                    fullName,
                     organization,
                     },
                 },
             } = this;
         const {
             latestRun: { id, result, startTime, endTime, changeSet, state, commitId, estimatedDurationInMillis },
             weatherScore,
-            name,
+            name: branchName,
         } = data;
-        const url = `/organizations/${organization}/${pipelineName}/detail/${name}/${id}/pipeline`;
+
+        const cleanBranchName = decodeURIComponent(branchName);
+        const url = buildRunDetailsUrl(organization, fullName, cleanBranchName, id, 'pipeline');
```

Comment:
Yes because we don't necessarily want to pass the encoded value around, in my opinion at least. Once we start to leverage the HAL stuff we shouldn't ever have to do any manual encoding, maybe decoding in select places.